### DISCUSSION TITLE
Got async warning while was implementing TCA on my project (Version: 0.33.0)

### DISCUSSION THREAD

**Discussion by BrunoCerberus** [pointfreeco/swift-composable-architecture#992] [Q&A]
<img width="1421" alt="Screen Shot 2022-02-04 at 17 46 00" src="https://user-images.githubusercontent.com/10541956/152602515-fb5f03ea-f7aa-4724-b0f8-bf43e60ae0e0.png">

**Comment by stephencelis** [comment]
Hi @BrunoCerberus! TCA's `Store` type is deliberately _not_ thread-safe. We mention this in the README's FAQ here:

https://github.com/pointfreeco/swift-composable-architecture#faq

And also [in the documentation for `Store`](https://pointfreeco.github.io/swift-composable-architecture/Store/). Here's a permalink to the specific code comment:

https://github.com/pointfreeco/swift-composable-architecture/blob/dd084dc147dadede0ad25f829a43ceafed2b9abc/Sources/ComposableArchitecture/Store.swift#L80

If you have more information, we can try to help you debug the problem further. It would be particularly helpful if you can attach a project that reproduces the problem you're seeing.
